packages/create-cloudflare/src/cli.ts

```diff
@@ -25,7 +26,13 @@ export const main = async (argv: string[]) => {
 	// Print a newline
 	logRaw("");
 
-	if (args.autoUpdate && (await isUpdateAvailable())) {
+	if (
+		args.autoUpdate &&
+		!process.env.VITEST &&
+		!process.env.CI &&
+		isInteractive() &&
```

Comment:
I moved these checks here to make it easier to test the `isUpdateAvailable()` function but also because it is not really nice having checks like that hidden deep down in the code.